tests/utils/test_scripts.py

```diff
@@ -158,3 +159,72 @@ def test_export_predictions_and_labeled_video(
 
     # clean up logging
     remove_logs()
+
+
+def _supervised_multi_gpu_cfg(cfg):
+    return OmegaConf.merge(
+        cfg,
+        OmegaConf.create(
+            {
+                "model": {
+                    "losses_to_use": [],
+                },
+                "training": {
+                    "num_gpus": 2,
+                    "train_batch_size": 4,
+                    "val_batch_size": 16,
+                    "test_batch_size": 16,
+                },
+            }
+        ),
+    )
+
+
+def test_get_data_module_num_gpus_0(cfg):
+    cfg = _supervised_multi_gpu_cfg(cfg)
+    # when num_gpus is set to 0, i.e. from a deprecated config
+    cfg.training.num_gpus = 0
+    data_module = get_data_module(cfg, Mock(spec=BaseTrackingDataset))
+
+    # assert num_gpus gets modified to 1
+    assert cfg.training.num_gpus == 1
+    # the rest of the behavior follows correctly
+    assert data_module.train_batch_size == 4
+    assert data_module.val_batch_size == 16
+    assert data_module.test_batch_size == 16
+
+
+def test_get_data_module_multi_gpu_batch_size_adjustment(cfg):
+    cfg = _supervised_multi_gpu_cfg(cfg)
+    data_module = get_data_module(cfg, Mock(spec=BaseTrackingDataset))
+    # train, val batch sizes should be divided by num_gpus
+    assert data_module.train_batch_size == 2
```

Comment:
same comment as above, pull `train_batch_size` and `num_gpus` out of cfg

Reply:
Done